Cajita doesn't build with gcc-10

Cajita doesn't build when kokkos is compile with CXX 11:
```
/home/junghans/Cabana/cajita/src/Cajita_GlobalMesh.hpp: In constructor ‘Cajita::GlobalMesh<MeshType>::GlobalMesh(const std::array<typename MeshType::scalar_type, 3>&, const std::array<typename MeshType::scalar_type, 3>&, const std::array<typename MeshType::scalar_type, 3>&)’:
/home/junghans/Cabana/cajita/src/Cajita_GlobalMesh.hpp:82:28: error: ‘logic_error’ is not a member of ‘std’
   82 |                 throw std::logic_error(
      |                            ^~~~~~~~~~~
/home/junghans/Cabana/cajita/src/Cajita_GlobalMesh.hpp: In constructor ‘Cajita::GlobalMesh<MeshType>::GlobalMesh(const std::array<typename MeshType::scalar_type, 3>&, const std::array<typename MeshType::scalar_type, 3>&, const std::array<int, 3>&)’:
/home/junghans/Cabana/cajita/src/Cajita_GlobalMesh.hpp:108:28: error: ‘logic_error’ is not a member of ‘std’
  108 |                 throw std::logic_error(
      |                            ^~~~~~~~~~~
/home/junghans/Cabana/cajita/src/Cajita_GlobalMesh.hpp:111:28: error: ‘logic_error’ is not a member of ‘std’
  111 |                 throw std::logic_error( "Global number of cells mismatch" );
      |                            ^~~~~~~~~~~
make[2]: *** [cajita/src/CMakeFiles/Cajita.dir/build.make:83: cajita/src/CMakeFiles/Cajita.dir/Cajita_LocalGrid.cpp.o] Error 1
```

We need to check the kokkos CXX standard somewhere or set it to C++14 independently.

--- a/cajita/src/Cajita_GlobalMesh.hpp
+++ b/cajita/src/Cajita_GlobalMesh.hpp
@@ -18,6 +18,7 @@
 #include <cmath>
 #include <limits>
 #include <memory>
+#include <stdexcept>
 #include <vector>
 
 namespace Cajita

